app/main.py

```diff
@@ -1,6 +1,16 @@
-from typing import Callable
+from typing import Callable, Any
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    history_def = {}
+
+    def wrapper(*args: Any) -> Callable:
+        if args not in history_def:
+            print("Calculating new result")
+            history_def[args] = func(*args)
+            return history_def[args]
+
+        print("Getting from cache")
+        return history_def[args]
```

Comment:
```suggestion
        else:
             print("Getting from cache")
        return history_def[args]
```
In this case better to use if-else because return the same